### PR TITLE
feat: SplitButton support

### LINKED ISSUE #2884: WinUI SplitButton
## What would you like to be added:

Support for SplitButton.

## Why is this needed:

WinUI parity.

## For which Platform:

- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #2884 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

No SplitButton.

## What is the new behavior?

Implemented SplitButton.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/SplitButtonEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.UI.Xaml.Controls;
+
+using SplitButton = Microsoft.UI.Xaml.Controls.SplitButton;
```
Comment (jeromelaban) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/SplitButtonEventArgs.cs]: Why did you need to add those namespaces ?

Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/SplitButtonEventArgs.cs]: Yes, it wasn't necessary here :-) .

**Comment by MartinZikmund** [comment]
> Nice stuff!
> 
> Would be awesome if you could add an automated test or two ;)

Will try :-)

**Comment by jeromelaban** [comment]
> > Would be awesome if you could add an automated test or two ;)
> 
> Will try :-)

You can most likely take the UI tests that are in WinUI, even if you have to adjust the method names.

**Comment by MartinZikmund** [comment]
@jeromelaban I have added the WinUI API test, but I am struggling with interaction tests - SplitButton behavior differs when used with touch vs mouse, so using `_app.Tap();` is not sufficient, and I need to be able to perform left and right click with mouse as well. Is there a workaround?

**Comment by jeromelaban** [comment]
> @jeromelaban I have added the WinUI API test, but I am struggling with interaction tests - SplitButton behavior differs when used with touch vs mouse, so using `_app.Tap();` is not sufficient, and I need to be able to perform left and right click with mouse as well. Is there a workaround?

The only way at this point to simulate a right click is to use the `LongPress`. Mouse vs. Touch is not possible at this point, but we'll have to add it soon for this type of scenarios.

**Comment by jeromelaban** [comment]
If porting the tests is not really applicable, you can add some simple tests to validate the behaviors you've discovered.

**Comment by MartinZikmund** [comment]
> If porting the tests is not really applicable, you can add some simple tests to validate the behaviors you've discovered.

Yes, will do asap

**Comment by MartinZikmund** [comment]
I found out there is a problem with #3165 which makes `SplitButton` not work properly the first time it is interacted with. This PR can probably be merged before then, but fixing the issue can make the control fully reliable :-) .

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/SplitButtonEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Microsoft.UI.Xaml.Controls
+{
+	public class SplitButtonClickEventArgs
```
Comment (dr1rrb) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/SplitButtonEventArgs.cs:3]: The ctor is not public on UWP, please make it internal

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/SplitButtonEventArgs.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+	{
+	}
+
+	public class ToggleSplitButtonIsCheckedChangedEventArgs
```
Comment (dr1rrb) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/SplitButtonEventArgs.cs:7]: Same thing, the ctor is not public on UWP, please make it internal

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/SplitButton.cs [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+			m_primaryButton = this.GetTemplateChild("PrimaryButton") as Button;
+			m_secondaryButton = this.GetTemplateChild("SecondaryButton") as Button;
+
+			var primaryButton = m_primaryButton;
+			if (primaryButton != null)
+			{
+				primaryButton.Click += OnClickPrimary;
+
+				m_pressedPrimaryRevoker = primaryButton.RegisterPropertyChangedCallback(ButtonBase.IsPressedProperty, OnVisualPropertyChanged);
+				m_pointerOverPrimaryRevoker = primaryButton.RegisterPropertyChangedCallback(ButtonBase.IsPointerOverProperty, OnVisualPropertyChanged);
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/SplitButton.cs:64]: Is this a known issue?

[the remark this answers is not in the document]
Comment (dr1rrb) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/SplitButton.cs:64]: Well now it is : https://github.com/unoplatform/uno/issues/3168 :)

Actually I discovered those properties (`IsPointerPressedPreoperty` and `IsPointerOverPreoperty`)  when I read your code. I was about to tell that both are not supported until I looked in Uno's source code and discovered that the support for the `IsPointerPressedPreoperty` was added recently by @jeromelaban (https://github.com/unoplatform/uno/pull/2290)

[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/SplitButton.cs:64]: Thanks!
